Update dependencies in vendor

kvch opened the pull request:
This PR aims to consolidate dependencies in order to migration to go modules.
It removes unused dependencies, updates packages with inconsistent versions and updates unsupported deps.

Updated:

| Name | From | To |
| --- | --- | --- |
| `4d63.com/tz` | 2019-08-31T23:37:02Z | 2019-11-24T06:07:01Z |
| `github.com/Azure/azure-sdk-for-go` | 2019-07-30T02:56:09Z | 2019-08-05T17:15:50Z |
| `github.com/awslabs/goformation` | 2018-10-29T15:41:58Z | v4.1.0 |
| `github.com/coreos/go-systemd` | v18 | v20 |
| `github.com/docker/docker` | v19.03.2 | v19.03.5 |
| `github.com/docker/distribution` | 2017-05-24T20:58:24Z | 2019-09-03T18:54:37Z |
| `github.com/elastic/go-perf` | 2019-08-22T17:42:12Z | 2019-12-12T13:55:12Z |
| `github.com/opencontainers/go-digest` | 2017-05-10T16:33:54Z | 2019-02-28T22:06:55Z |
| `github.com/opencontainers/image-spec` | 2017-05-25T20:40:40Z | 2019-08-23T10:51:29Z |
| `github.com/prometheus/client_golang/prometheus` | 2019-08-20T08:35:54Z | 2019-09-13T10:31:02Z |
| `github.com/prometheus/procfs` | 2018-03-10T14:15:09Z | 2019-12-02T18:22:43Z |
| `golang.org/x/crypto` | 2018-09-26T22:02:31Z | release-branch.go1.13 |
| `golang.org/x/net` | release-branch.go1.9 | release-branch.go1.13 |
| `golang.org/x/sys` | 2019-07-11T09:16:12Z | release-branch.go1.13 |
| `golang.org/x/text` | 2018-12-15T14:04:16Z | release-branch.go1.13 |
| `golang.org/x/time/rate` | 2018-03-14T17:21:34Z | release-branch.go1.13 |
| `golang.org/x/tools` | release-branch.go1.10 | release-branch.go1.13 |
| `google.golang.org/genproto` | 2019-06-11T19:02:12Z | 2019-08-01T16:59:51Z |
| `google.golang.org/grpc` | v1.21.1 | v1.25.1 |

Moved from local vendor to global one:
- `github.com/Azure/go-ansiterm`
- `github.com/containerd/containerd/errdefs`
- `github.com/containerd/fifo`
- `github.com/coreos/go-systemd/activation`
- `github.com/davecgh/go-xdr/xdr2`
- `github.com/digitalocean/go-libvirt`
- `github.com/docker/go-plugins-helpers/sdk`
- `github.com/go-sql-driver/mysql`
- `github.com/google/uuid`
- `github.com/jmoiron/sqlx`
- `github.com/lib/pq`
- `github.com/morikuni/aec`
- `github.com/shirou/gopsutil`
- `github.com/vmware/govmomi`

Follow-ups:
- With the new verison of `goimports`, imports are changed: https://github.com/elastic/beats/pull/15231/commits/6ab6ccf7d6c24bc3392613ee74755312c430f449
- `.` is no longer an accepted domain in `registered_domain` due to https://github.com/golang/net/commit/afe8014c977f78fe2406743d9d32c8bd44d0542b#diff-a996f73c6de010758e5cd328723ed5a7: https://github.com/elastic/beats/pull/15231/files#diff-1e6d077db00bd53d2219274e906abc75
- refactored `accessibleProcs` as `fs.FS` was exported anymore in https://github.com/prometheus/procfs/commit/740c0778500738e56e1877d52904c431a3fdc573#diff-66272a983b0d81466c33c6e80936fc5b: https://github.com/elastic/beats/pull/15231/files#diff-feda550dd047db46b0030a5cd24f464f

faec commented:
I notice a lot of `vendor.json` entries remove the semantic version field they used to have and now exclusively use `revision` with a hash... is this intended?

jsoriano commented on vendor/vendor.json:
```diff
@@ -2796,33 +2794,73 @@
 			"revision": "fc7e5e7af6052e36e83e5539148015ed2c09d8f9",
-			"revisionTime": "2019-11-27T18:28:36Z",
-			"version": "v2.19.11",
-			"versionExact": "v2.19.11"
```
Could we continue using tagged versions when possible?

kvch replied:
Sure.

jsoriano commented on vendor/vendor.json:
```diff
@@ -3377,50 +3413,40 @@
-			"revision": "44b7c21cbf19450f38b337eb6b6fe4f6496fb5b3",
-			"revisionTime": "2017-10-22T17:59:00Z",
-			"version": "release-branch.go1.9",
-			"versionExact": "release-branch.go1.9"
```
Also in these libraries could we use the release branch?

urso commented:
> Should we move these dependencies to the main vendor directory to avoid problems with go modules?

If it doesn't break anything, yes please. We can do so in a separate PR. Look out for versions or potential incompatibilities. When switching to go-modules we will get rid of local vendors.

urso commented:
On Jenkins in two cases exactly the same test did fail. Looks like a bug or faulty test. Could this be related to this change or a problem with the code/test in general?

ph commented:
good catch @faec

kvch commented:
@faec @ph Yes, it was intentional. I have selected the latest version from each dependency. If the latest did not have an exact version, I went with the revision.

@urso I will add those deps to this PR. Also, I am looking into the CI fails.

For everything under `golang.org/x`, I am fetching the release branch for go 1.13. I think we should update these deps whenever we update the supported go version of Beats. Thus, when we upgrade to go 1.14, we should update these deps as well. 

Thank you everyone for the review notes!

urso commented:
> For everything under golang.org/x, I am fetching the release branch for go 1.13. I think we should update these deps whenever we update the supported go version of Beats. Thus, when we upgrade to go 1.14, we should update these deps as well.

+1

narph commented:
> Change LGTM, but I would like @narph to double check for the Azure dependencies to make sure we don't break anything.

LGTM

kvch commented:
@urso @jsoriano I have moved deps from local vendors and updated the PR description with the version changes. Please also check the files where I had to follow up changes.

urso commented:
@kvch @fearful-symmetry Looks like dockerlogbeat is still broken on travis. I didn't see dockerlogbeat being tested on jenkins.

@kvch @andrewkroh 19 winlogbeat tests are failing (all in `TestSecurity`). Are these cause by the update or is it some known flakiness due to the windows version they are run in?

kvch commented:
@urso it seems related to the update. I am looking into it.

kvch commented:
Winlogbeat errors were not related. Those tests are green now.

kvch commented:
Travis is green! \o/

kvch commented:
jenkins test this

kvch commented:
Everything is green. \o/ \o/ \o/
